2024-01-14 10:54:58.417 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 10:54:58.418 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 10:54:58.418 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 10:54:58.722 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 10:54:58.852 | INFO     | logging:callHandlers:1706 - Started server process [12892]
2024-01-14 10:54:58.852 | INFO     | logging:callHandlers:1706 - Started server process [12892]
2024-01-14 10:54:58.853 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 10:54:58.853 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 10:54:58.885 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 10:54:58.885 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 10:55:03.355 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52246 - "POST /users/login HTTP/1.1" 405
2024-01-14 10:55:17.229 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52256 - "GET /users/login HTTP/1.1" 401
2024-01-14 10:55:38.975 | DEBUG    | logging:callHandlers:1706 - Calling on_part_begin with no data
2024-01-14 10:55:38.975 | DEBUG    | logging:callHandlers:1706 - Calling on_header_field with data[54:73]
2024-01-14 10:55:38.976 | DEBUG    | logging:callHandlers:1706 - Calling on_header_value with data[75:98]
2024-01-14 10:55:38.976 | DEBUG    | logging:callHandlers:1706 - Calling on_header_end with no data
2024-01-14 10:55:38.977 | DEBUG    | logging:callHandlers:1706 - Calling on_headers_finished with no data
2024-01-14 10:55:38.977 | DEBUG    | logging:callHandlers:1706 - Calling on_part_data with data[102:116]
2024-01-14 10:55:38.978 | DEBUG    | logging:callHandlers:1706 - Calling on_part_end with no data
2024-01-14 10:55:38.978 | DEBUG    | logging:callHandlers:1706 - Calling on_part_begin with no data
2024-01-14 10:55:38.978 | DEBUG    | logging:callHandlers:1706 - Calling on_header_field with data[172:191]
2024-01-14 10:55:38.979 | DEBUG    | logging:callHandlers:1706 - Calling on_header_value with data[193:219]
2024-01-14 10:55:38.979 | DEBUG    | logging:callHandlers:1706 - Calling on_header_end with no data
2024-01-14 10:55:38.979 | DEBUG    | logging:callHandlers:1706 - Calling on_headers_finished with no data
2024-01-14 10:55:38.980 | DEBUG    | logging:callHandlers:1706 - Calling on_part_data with data[223:224]
2024-01-14 10:55:38.980 | DEBUG    | logging:callHandlers:1706 - Calling on_part_end with no data
2024-01-14 10:55:38.981 | DEBUG    | logging:callHandlers:1706 - Calling on_end with no data
2024-01-14 10:55:39.001 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52258 - "POST /auth/login HTTP/1.1" 422
2024-01-14 10:55:49.162 | DEBUG    | logging:callHandlers:1706 - Calling on_part_begin with no data
2024-01-14 10:55:49.162 | DEBUG    | logging:callHandlers:1706 - Calling on_header_field with data[54:73]
2024-01-14 10:55:49.162 | DEBUG    | logging:callHandlers:1706 - Calling on_header_value with data[75:100]
2024-01-14 10:55:49.163 | DEBUG    | logging:callHandlers:1706 - Calling on_header_end with no data
2024-01-14 10:55:49.163 | DEBUG    | logging:callHandlers:1706 - Calling on_headers_finished with no data
2024-01-14 10:55:49.163 | DEBUG    | logging:callHandlers:1706 - Calling on_part_data with data[104:118]
2024-01-14 10:55:49.164 | DEBUG    | logging:callHandlers:1706 - Calling on_part_end with no data
2024-01-14 10:55:49.164 | DEBUG    | logging:callHandlers:1706 - Calling on_part_begin with no data
2024-01-14 10:55:49.164 | DEBUG    | logging:callHandlers:1706 - Calling on_header_field with data[174:193]
2024-01-14 10:55:49.165 | DEBUG    | logging:callHandlers:1706 - Calling on_header_value with data[195:221]
2024-01-14 10:55:49.165 | DEBUG    | logging:callHandlers:1706 - Calling on_header_end with no data
2024-01-14 10:55:49.165 | DEBUG    | logging:callHandlers:1706 - Calling on_headers_finished with no data
2024-01-14 10:55:49.165 | DEBUG    | logging:callHandlers:1706 - Calling on_part_data with data[225:226]
2024-01-14 10:55:49.166 | DEBUG    | logging:callHandlers:1706 - Calling on_part_end with no data
2024-01-14 10:55:49.166 | DEBUG    | logging:callHandlers:1706 - Calling on_end with no data
2024-01-14 10:55:49.168 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52264 - "POST /auth/login HTTP/1.1" 422
2024-01-14 10:55:55.937 | DEBUG    | logging:callHandlers:1706 - Calling on_part_begin with no data
2024-01-14 10:55:55.937 | DEBUG    | logging:callHandlers:1706 - Calling on_header_field with data[54:73]
2024-01-14 10:55:55.938 | DEBUG    | logging:callHandlers:1706 - Calling on_header_value with data[75:101]
2024-01-14 10:55:55.939 | DEBUG    | logging:callHandlers:1706 - Calling on_header_end with no data
2024-01-14 10:55:55.939 | DEBUG    | logging:callHandlers:1706 - Calling on_headers_finished with no data
2024-01-14 10:55:55.940 | DEBUG    | logging:callHandlers:1706 - Calling on_part_data with data[105:119]
2024-01-14 10:55:55.940 | DEBUG    | logging:callHandlers:1706 - Calling on_part_end with no data
2024-01-14 10:55:55.941 | DEBUG    | logging:callHandlers:1706 - Calling on_part_begin with no data
2024-01-14 10:55:55.941 | DEBUG    | logging:callHandlers:1706 - Calling on_header_field with data[175:194]
2024-01-14 10:55:55.941 | DEBUG    | logging:callHandlers:1706 - Calling on_header_value with data[196:222]
2024-01-14 10:55:55.942 | DEBUG    | logging:callHandlers:1706 - Calling on_header_end with no data
2024-01-14 10:55:55.942 | DEBUG    | logging:callHandlers:1706 - Calling on_headers_finished with no data
2024-01-14 10:55:55.942 | DEBUG    | logging:callHandlers:1706 - Calling on_part_data with data[226:227]
2024-01-14 10:55:55.943 | DEBUG    | logging:callHandlers:1706 - Calling on_part_end with no data
2024-01-14 10:55:55.943 | DEBUG    | logging:callHandlers:1706 - Calling on_end with no data
2024-01-14 10:55:56.325 | WARNING  | logging:callHandlers:1706 - (trapped) error reading bcrypt versionTraceback (most recent call last):

  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 995, in _bootstrap
    self._bootstrap_inner()
    │    └ <function Thread._bootstrap_inner at 0x000001CB5B641C60>
    └ <WorkerThread(AnyIO worker thread, started 1944)>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\threading.py", line 1038, in _bootstrap_inner
    self.run()
    │    └ <function WorkerThread.run at 0x000001CB615A9F80>
    └ <WorkerThread(AnyIO worker thread, started 1944)>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
             │       │   │      └ ()
             │       │   └ functools.partial(<function login_access_token at 0x000001CB61275800>, form_data=<fastapi.security.oauth2.OAuth2PasswordReque...
             │       └ <method 'run' of '_contextvars.Context' objects>
             └ <_contextvars.Context object at 0x000001CB61575F00>

  File "E:\spending-webapp\backend\app\shared\decorator.py", line 24, in wrapper
    response = f(*args, **kwargs)
               │  │       └ {'form_data': <fastapi.security.oauth2.OAuth2PasswordRequestForm object at 0x000001CB615A3A50>, 'login_use_case': <app.use_ca...
               │  └ ()
               └ <function login_access_token at 0x000001CB61275760>

  File "E:\spending-webapp\backend\app\interfaces\rest\api_v1\endpoints\auth.py", line 42, in login_access_token
    response = login_use_case.execute(request_object=login_request_object)
               │              │                      └ <app.use_cases.auth.login.LoginRequestObject object at 0x000001CB608C0910>
               │              └ <function UseCase.execute at 0x000001CB61274FE0>
               └ <app.use_cases.auth.login.LoginUseCase object at 0x000001CB6129B4D0>

  File "E:\spending-webapp\backend\app\shared\use_case.py", line 30, in execute
    result = self.process_request(request_object)
             │    │               └ <app.use_cases.auth.login.LoginRequestObject object at 0x000001CB608C0910>
             │    └ <function LoginUseCase.process_request at 0x000001CB61275260>
             └ <app.use_cases.auth.login.LoginUseCase object at 0x000001CB6129B4D0>

  File "E:\spending-webapp\backend\app\use_cases\auth\login.py", line 47, in process_request
    user: UserModel = self.security_service.authenticate_user(
                      │    │                └ <function SecurityService.authenticate_user at 0x000001CB61274C20>
                      │    └ <app.infra.security.security_service.SecurityService object at 0x000001CB6126C750>
                      └ <app.use_cases.auth.login.LoginUseCase object at 0x000001CB6129B4D0>

  File "E:\spending-webapp\backend\app\infra\security\security_service.py", line 160, in authenticate_user
    if not verify_password(password, user.hashed_password):
           │               │         │    └ <mongoengine.fields.StringField object at 0x000001CB609BB890>
           │               │         └ <User: User object>
           │               └ '1'
           └ <function verify_password at 0x000001CB61274360>

  File "E:\spending-webapp\backend\app\infra\security\security_service.py", line 33, in verify_password
    return pwd_context.verify(plain_password, hashed_password)
           │           │      │               └ '$2b$12$5ue7GYek4lNLwIkrS/0G7eF5pNp88/3Z0WyOU1FVeA04YUnbxXWi2'
           │           │      └ '1'
           │           └ <function CryptContext.verify at 0x000001CB60C722A0>
           └ <CryptContext at 0x1cb60a1c150>

  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\context.py", line 2347, in verify
    return record.verify(secret, hash, **kwds)
           │      │      │       │       └ {}
           │      │      │       └ '$2b$12$5ue7GYek4lNLwIkrS/0G7eF5pNp88/3Z0WyOU1FVeA04YUnbxXWi2'
           │      │      └ '1'
           │      └ <classmethod(<function GenericHandler.verify at 0x000001CB60C1FA60>)>
           └ <class 'passlib.handlers.bcrypt.<customized bcrypt hasher>'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\utils\handlers.py", line 792, in verify
    return consteq(self._calc_checksum(secret), chk)
           │       │    │              │        └ 'F5pNp88/3Z0WyOU1FVeA04YUnbxXWi2'
           │       │    │              └ '1'
           │       │    └ <function _NoBackend._calc_checksum at 0x000001CB6124B1A0>
           │       └ <passlib.handlers.bcrypt.<customized bcrypt hasher> object at 0x000001CB612F5110>
           └ <built-in function compare_digest>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\handlers\bcrypt.py", line 591, in _calc_checksum
    self._stub_requires_backend()
    │    └ <classmethod(<function BackendMixin._stub_requires_backend at 0x000001CB60C3DDA0>)>
    └ <passlib.handlers.bcrypt.<customized bcrypt hasher> object at 0x000001CB612F5110>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\utils\handlers.py", line 2254, in _stub_requires_backend
    cls.set_backend()
    │   └ <classmethod(<function BackendMixin.set_backend at 0x000001CB60C3DB20>)>
    └ <class 'passlib.handlers.bcrypt.<customized bcrypt hasher>'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\utils\handlers.py", line 2156, in set_backend
    return owner.set_backend(name, dryrun=dryrun)
           │     │           │            └ False
           │     │           └ 'any'
           │     └ <classmethod(<function BackendMixin.set_backend at 0x000001CB60C3DB20>)>
           └ <class 'passlib.handlers.bcrypt.bcrypt'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\utils\handlers.py", line 2163, in set_backend
    return cls.set_backend(name, dryrun=dryrun)
           │   │           │            └ False
           │   │           └ 'bcrypt'
           │   └ <classmethod(<function BackendMixin.set_backend at 0x000001CB60C3DB20>)>
           └ <class 'passlib.handlers.bcrypt.bcrypt'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\utils\handlers.py", line 2188, in set_backend
    cls._set_backend(name, dryrun)
    │   │            │     └ False
    │   │            └ 'bcrypt'
    │   └ <classmethod(<function SubclassBackendMixin._set_backend at 0x000001CB60C3DEE0>)>
    └ <class 'passlib.handlers.bcrypt.bcrypt'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\utils\handlers.py", line 2311, in _set_backend
    super(SubclassBackendMixin, cls)._set_backend(name, dryrun)
          │                     │                 │     └ False
          │                     │                 └ 'bcrypt'
          │                     └ <class 'passlib.handlers.bcrypt.bcrypt'>
          └ <class 'passlib.utils.handlers.SubclassBackendMixin'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\utils\handlers.py", line 2224, in _set_backend
    ok = loader(**kwds)
         │        └ {'name': 'bcrypt', 'dryrun': False}
         └ <bound method _BcryptBackend._load_backend_mixin of <class 'passlib.handlers.bcrypt._BcryptBackend'>>
> File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\passlib\handlers\bcrypt.py", line 620, in _load_backend_mixin
    version = _bcrypt.__about__.__version__
              └ <module 'bcrypt' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\bcrypt\\__init__.py'>

AttributeError: module 'bcrypt' has no attribute '__about__'

2024-01-14 10:55:56.364 | DEBUG    | logging:callHandlers:1706 - detected 'bcrypt' backend, version '<unknown>'
2024-01-14 10:55:56.364 | DEBUG    | logging:callHandlers:1706 - 'bcrypt' backend lacks $2$ support, enabling workaround
2024-01-14 10:55:57.349 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52266 - "POST /auth/login HTTP/1.1" 200
2024-01-14 10:58:11.017 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 10:58:11.017 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 10:58:11.127 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 10:58:11.127 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 10:58:11.129 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 10:58:11.129 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 10:58:11.129 | INFO     | logging:callHandlers:1706 - Finished server process [12892]
2024-01-14 10:58:11.129 | INFO     | logging:callHandlers:1706 - Finished server process [12892]
2024-01-14 10:58:12.598 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 10:58:12.598 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 10:58:12.599 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 10:58:12.744 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 10:58:12.837 | INFO     | logging:callHandlers:1706 - Started server process [9348]
2024-01-14 10:58:12.838 | INFO     | logging:callHandlers:1706 - Started server process [9348]
2024-01-14 10:58:12.838 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 10:58:12.838 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 10:58:12.846 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 10:58:12.846 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 10:58:32.369 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52286 - "POST /category HTTP/1.1" 404
2024-01-14 10:58:43.919 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52292 - "POST /categories HTTP/1.1" 401
2024-01-14 10:59:33.100 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52364 - "POST /categories HTTP/1.1" 500
2024-01-14 11:04:45.966 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:04:45.966 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:04:46.075 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:04:46.075 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:04:46.077 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:04:46.078 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:04:46.078 | INFO     | logging:callHandlers:1706 - Finished server process [9348]
2024-01-14 11:04:46.079 | INFO     | logging:callHandlers:1706 - Finished server process [9348]
2024-01-14 11:04:47.566 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:04:47.566 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:04:47.567 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:04:47.714 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:04:47.808 | INFO     | logging:callHandlers:1706 - Started server process [10708]
2024-01-14 11:04:47.808 | INFO     | logging:callHandlers:1706 - Started server process [10708]
2024-01-14 11:04:47.809 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:04:47.810 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:04:47.817 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:04:47.817 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:05:29.719 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52424 - "POST /categories HTTP/1.1" 500
2024-01-14 11:08:36.286 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:08:36.286 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:08:36.396 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:08:36.396 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:08:36.398 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:08:36.398 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:08:36.398 | INFO     | logging:callHandlers:1706 - Finished server process [10708]
2024-01-14 11:08:36.399 | INFO     | logging:callHandlers:1706 - Finished server process [10708]
2024-01-14 11:08:37.534 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:08:37.535 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:08:37.536 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:08:37.673 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:08:37.769 | INFO     | logging:callHandlers:1706 - Started server process [3032]
2024-01-14 11:08:37.770 | INFO     | logging:callHandlers:1706 - Started server process [3032]
2024-01-14 11:08:37.770 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:08:37.770 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:08:37.778 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:08:37.778 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:10:06.845 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:10:06.845 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:10:06.953 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:10:06.953 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:10:06.954 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:10:06.955 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:10:06.955 | INFO     | logging:callHandlers:1706 - Finished server process [3032]
2024-01-14 11:10:06.955 | INFO     | logging:callHandlers:1706 - Finished server process [3032]
2024-01-14 11:10:08.491 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:10:08.491 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:10:08.492 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:10:08.633 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:10:18.941 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:10:18.942 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:10:18.942 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:10:19.084 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:11:17.286 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:11:17.286 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:11:17.287 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:11:17.426 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:11:17.519 | INFO     | logging:callHandlers:1706 - Started server process [4204]
2024-01-14 11:11:17.519 | INFO     | logging:callHandlers:1706 - Started server process [4204]
2024-01-14 11:11:17.520 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:11:17.520 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:11:17.527 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:11:17.527 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:11:20.446 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52477 - "POST /categories HTTP/1.1" 500
2024-01-14 11:11:20.447 | ERROR    | logging:callHandlers:1706 - Exception in ASGI application
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 308
               │     └ 3
               └ <function _main at 0x0000018F32A67CE0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 135, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 308
           │    └ <function BaseProcess._bootstrap at 0x0000018F3275C7C0>
           └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000018F3275BCE0>
    └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000018F32A2FB50>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
    │    └ <function subprocess_started at 0x0000018F34D8FCE0>
    └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=768, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000018F34EADD90>>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=768, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
           │       │   │    └ <function Server.serve at 0x0000018F34D8ED40>
           │       │   └ <uvicorn.server.Server object at 0x0000018F34EADD90>
           │       └ <function run at 0x0000018F34618220>
           └ <module 'asyncio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000018F34EB4AE0>
           │      └ <function Runner.run at 0x0000018F34694180>
           └ <asyncio.runners.Runner object at 0x0000018F32A2FAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-p...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000018F3468DC60>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000018F32A2FAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000018F3468DBC0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000018F3468F9C0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1922, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000018F345D5260>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
> File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000018F38361610>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018F38...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000018F350765D0>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000018F38361610>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018F38...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018F38...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000018F38499410>
          └ <fastapi.applications.FastAPI object at 0x0000018F350765D0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000018F384DD4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000018F383A64D0>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000018F38499410>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\cors.py", line 83, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000018F384DD4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.sessions.SessionMiddleware object at 0x0000018F38500450>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000018F383A64D0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\sessions.py", line 86, in __call__
    await self.app(scope, receive, send_wrapper)
          │    │   │      │        └ <function SessionMiddleware.__call__.<locals>.send_wrapper at 0x0000018F384DE980>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000018F38500710>
          └ <starlette.middleware.sessions.SessionMiddleware object at 0x0000018F38500450>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 79, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000018F385007D0>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000018F38500710>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 20, in __call__
    raise e
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000018F3506D650>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000018F385007D0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x0000018F36F3F600>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000018F38416C00>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000018F38503E50>
                     └ <function get_request_handler.<locals>.app at 0x0000018F38417B00>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000018F36F3D940>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 193, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
                 │                 │         │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
                 │                 │         └ <function create_category at 0x0000018F38336480>
                 │                 └ <fastapi.dependencies.models.Dependant object at 0x0000018F384DA210>
                 └ <function run_in_threadpool at 0x0000018F35C080E0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
                 │     │         │        │      └ ()
                 │     │         │        └ functools.partial(<function create_category at 0x0000018F38336480>, current_user=<User: User object>, create_category_use_cas...
                 │     │         └ <function run_sync at 0x0000018F34E11080>
                 │     └ <module 'anyio.to_thread' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\to_...
                 └ <module 'anyio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
                 └ <function get_asynclib at 0x0000018F34DD2200>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
                 └ <Future finished exception=TypeError("CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
             │       │   │      └ ()
             │       │   └ functools.partial(<function create_category at 0x0000018F38336480>, current_user=<User: User object>, create_category_use_cas...
             │       └ <method 'run' of '_contextvars.Context' objects>
             └ <_contextvars.Context object at 0x0000018F38437FC0>

  File "E:\spending-webapp\backend\app\shared\decorator.py", line 24, in wrapper
    response = f(*args, **kwargs)
               │  │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
               │  └ ()
               └ <function create_category at 0x0000018F383379C0>

  File "E:\spending-webapp\backend\app\interfaces\rest\api_v1\endpoints\category.py", line 44, in create_category
    req_object = CreateCategoryRequestObject.builder(payload=payload)
                 │                           │               └ CategoryInCreate(name='Lương tháng', type=<Type.INCOME: 'income'>, note='Dùng khi nhận lương')
                 │                           └ <classmethod(<function CreateCategoryRequestObject.builder at 0x0000018F38357920>)>
                 └ <class 'app.use_cases.category.create.CreateCategoryRequestObject'>

TypeError: CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_user'

2024-01-14 11:11:20.483 | ERROR    | logging:callHandlers:1706 - Exception in ASGI application
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 308
               │     └ 3
               └ <function _main at 0x0000018F32A67CE0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 135, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 308
           │    └ <function BaseProcess._bootstrap at 0x0000018F3275C7C0>
           └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000018F3275BCE0>
    └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x0000018F32A2FB50>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
    │    └ <function subprocess_started at 0x0000018F34D8FCE0>
    └ <SpawnProcess name='SpawnProcess-1' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=768, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x0000018F34EADD90>>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=768, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
           │       │   │    └ <function Server.serve at 0x0000018F34D8ED40>
           │       │   └ <uvicorn.server.Server object at 0x0000018F34EADD90>
           │       └ <function run at 0x0000018F34618220>
           └ <module 'asyncio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x0000018F34EB4AE0>
           │      └ <function Runner.run at 0x0000018F34694180>
           └ <asyncio.runners.Runner object at 0x0000018F32A2FAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-p...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x0000018F3468DC60>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x0000018F32A2FAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x0000018F3468DBC0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x0000018F3468F9C0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1922, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x0000018F345D5260>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
> File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000018F38361610>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018F38...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x0000018F350765D0>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x0000018F38361610>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018F38...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018F38...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000018F38499410>
          └ <fastapi.applications.FastAPI object at 0x0000018F350765D0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000018F384DD4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x0000018F383A64D0>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x0000018F38499410>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\cors.py", line 83, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x0000018F384DD4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.sessions.SessionMiddleware object at 0x0000018F38500450>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x0000018F383A64D0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\sessions.py", line 86, in __call__
    await self.app(scope, receive, send_wrapper)
          │    │   │      │        └ <function SessionMiddleware.__call__.<locals>.send_wrapper at 0x0000018F384DE980>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000018F38500710>
          └ <starlette.middleware.sessions.SessionMiddleware object at 0x0000018F38500450>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 79, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000018F385007D0>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x0000018F38500710>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 20, in __call__
    raise e
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x0000018F3506D650>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x0000018F385007D0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x0000018F36F3F600>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x0000018F384DEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000018...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x0000018F38416C00>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x0000018F38503E50>
                     └ <function get_request_handler.<locals>.app at 0x0000018F38417B00>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x0000018F36F3D940>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 193, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
                 │                 │         │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
                 │                 │         └ <function create_category at 0x0000018F38336480>
                 │                 └ <fastapi.dependencies.models.Dependant object at 0x0000018F384DA210>
                 └ <function run_in_threadpool at 0x0000018F35C080E0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
                 │     │         │        │      └ ()
                 │     │         │        └ functools.partial(<function create_category at 0x0000018F38336480>, current_user=<User: User object>, create_category_use_cas...
                 │     │         └ <function run_sync at 0x0000018F34E11080>
                 │     └ <module 'anyio.to_thread' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\to_...
                 └ <module 'anyio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
                 └ <function get_asynclib at 0x0000018F34DD2200>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
                 └ <Future finished exception=TypeError("CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
             │       │   │      └ ()
             │       │   └ functools.partial(<function create_category at 0x0000018F38336480>, current_user=<User: User object>, create_category_use_cas...
             │       └ <method 'run' of '_contextvars.Context' objects>
             └ <_contextvars.Context object at 0x0000018F38437FC0>

  File "E:\spending-webapp\backend\app\shared\decorator.py", line 24, in wrapper
    response = f(*args, **kwargs)
               │  │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
               │  └ ()
               └ <function create_category at 0x0000018F383379C0>

  File "E:\spending-webapp\backend\app\interfaces\rest\api_v1\endpoints\category.py", line 44, in create_category
    req_object = CreateCategoryRequestObject.builder(payload=payload)
                 │                           │               └ CategoryInCreate(name='Lương tháng', type=<Type.INCOME: 'income'>, note='Dùng khi nhận lương')
                 │                           └ <classmethod(<function CreateCategoryRequestObject.builder at 0x0000018F38357920>)>
                 └ <class 'app.use_cases.category.create.CreateCategoryRequestObject'>

TypeError: CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_user'

2024-01-14 11:11:54.149 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:11:54.149 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:11:54.257 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:11:54.257 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:11:54.259 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:11:54.259 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:11:54.259 | INFO     | logging:callHandlers:1706 - Finished server process [4204]
2024-01-14 11:11:54.260 | INFO     | logging:callHandlers:1706 - Finished server process [4204]
2024-01-14 11:11:55.727 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:11:55.728 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:11:55.728 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:11:55.878 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:11:55.974 | INFO     | logging:callHandlers:1706 - Started server process [11884]
2024-01-14 11:11:55.975 | INFO     | logging:callHandlers:1706 - Started server process [11884]
2024-01-14 11:11:55.976 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:11:55.976 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:11:55.983 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:11:55.984 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:11:59.008 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52493 - "POST /categories HTTP/1.1" 500
2024-01-14 11:11:59.008 | ERROR    | logging:callHandlers:1706 - Exception in ASGI application
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 308
               │     └ 3
               └ <function _main at 0x000001979A2F7CE0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 135, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 308
           │    └ <function BaseProcess._bootstrap at 0x0000019799FEC7C0>
           └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000019799FEBCE0>
    └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x000001979A2BFB50>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
    │    └ <function subprocess_started at 0x000001979C6AFCE0>
    └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=788, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x000001979C7DDD90>>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=788, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
           │       │   │    └ <function Server.serve at 0x000001979C6AED40>
           │       │   └ <uvicorn.server.Server object at 0x000001979C7DDD90>
           │       └ <function run at 0x000001979BEE8220>
           └ <module 'asyncio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x000001979C7E4AE0>
           │      └ <function Runner.run at 0x000001979BF64180>
           └ <asyncio.runners.Runner object at 0x000001979A2BFAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-p...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x000001979BF5DC60>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x000001979A2BFAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x000001979BF5DBC0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x000001979BF5F9C0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1922, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x000001979BE95260>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
> File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x000001979FBF2ED0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x000001979F...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x000001979E90FE90>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x000001979FBF2ED0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x000001979F...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x000001979F...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x000001979F3B0610>
          └ <fastapi.applications.FastAPI object at 0x000001979E90FE90>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x000001979FDED4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x000001979FC4E910>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x000001979F3B0610>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\cors.py", line 83, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x000001979FDED4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.sessions.SessionMiddleware object at 0x000001979FE10090>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x000001979FC4E910>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\sessions.py", line 86, in __call__
    await self.app(scope, receive, send_wrapper)
          │    │   │      │        └ <function SessionMiddleware.__call__.<locals>.send_wrapper at 0x000001979FDEE980>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x000001979FE10950>
          └ <starlette.middleware.sessions.SessionMiddleware object at 0x000001979FE10090>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 79, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x000001979FE10C50>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x000001979FE10950>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 20, in __call__
    raise e
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x000001979C9BD650>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x000001979FE10C50>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x000001979E88F600>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x000001979FD27E20>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x000001979FE13E50>
                     └ <function get_request_handler.<locals>.app at 0x000001979FD25440>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x000001979E88D940>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 193, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
                 │                 │         │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
                 │                 │         └ <function create_category at 0x000001979FC66AC0>
                 │                 └ <fastapi.dependencies.models.Dependant object at 0x000001979FDEA750>
                 └ <function run_in_threadpool at 0x000001979CD580E0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
                 │     │         │        │      └ ()
                 │     │         │        └ functools.partial(<function create_category at 0x000001979FC66AC0>, current_user=<User: User object>, create_category_use_cas...
                 │     │         └ <function run_sync at 0x000001979C741080>
                 │     └ <module 'anyio.to_thread' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\to_...
                 └ <module 'anyio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
                 └ <function get_asynclib at 0x000001979C6F2200>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
                 └ <Future finished exception=TypeError("CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
             │       │   │      └ ()
             │       │   └ functools.partial(<function create_category at 0x000001979FC66AC0>, current_user=<User: User object>, create_category_use_cas...
             │       └ <method 'run' of '_contextvars.Context' objects>
             └ <_contextvars.Context object at 0x000001979FF6F680>

  File "E:\spending-webapp\backend\app\shared\decorator.py", line 24, in wrapper
    response = f(*args, **kwargs)
               │  │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
               │  └ ()
               └ <function create_category at 0x000001979FC66A20>

  File "E:\spending-webapp\backend\app\interfaces\rest\api_v1\endpoints\category.py", line 44, in create_category
    req_object = CreateCategoryRequestObject.builder(payload=payload)
                 │                           │               └ CategoryInCreate(name='Lương tháng', type=<Type.INCOME: 'income'>, note='Dùng khi nhận lương')
                 │                           └ <classmethod(<function CreateCategoryRequestObject.builder at 0x000001979FC67920>)>
                 └ <class 'app.use_cases.category.create.CreateCategoryRequestObject'>

TypeError: CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_user'

2024-01-14 11:11:59.020 | ERROR    | logging:callHandlers:1706 - Exception in ASGI application
Traceback (most recent call last):

  File "<string>", line 1, in <module>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 122, in spawn_main
    exitcode = _main(fd, parent_sentinel)
               │     │   └ 308
               │     └ 3
               └ <function _main at 0x000001979A2F7CE0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\spawn.py", line 135, in _main
    return self._bootstrap(parent_sentinel)
           │    │          └ 308
           │    └ <function BaseProcess._bootstrap at 0x0000019799FEC7C0>
           └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 314, in _bootstrap
    self.run()
    │    └ <function BaseProcess.run at 0x0000019799FEBCE0>
    └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\multiprocessing\process.py", line 108, in run
    self._target(*self._args, **self._kwargs)
    │    │        │    │        │    └ {'config': <uvicorn.config.Config object at 0x000001979A2BFB50>, 'target': <bound method Server.run of <uvicorn.server.Server...
    │    │        │    │        └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
    │    │        │    └ ()
    │    │        └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
    │    └ <function subprocess_started at 0x000001979C6AFCE0>
    └ <SpawnProcess name='SpawnProcess-2' parent=11240 started>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\_subprocess.py", line 76, in subprocess_started
    target(sockets=sockets)
    │              └ [<socket.socket fd=788, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
    └ <bound method Server.run of <uvicorn.server.Server object at 0x000001979C7DDD90>>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\server.py", line 61, in run
    return asyncio.run(self.serve(sockets=sockets))
           │       │   │    │             └ [<socket.socket fd=788, family=2, type=1, proto=6, laddr=('127.0.0.1', 8000)>]
           │       │   │    └ <function Server.serve at 0x000001979C6AED40>
           │       │   └ <uvicorn.server.Server object at 0x000001979C7DDD90>
           │       └ <function run at 0x000001979BEE8220>
           └ <module 'asyncio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\asyncio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 190, in run
    return runner.run(main)
           │      │   └ <coroutine object Server.serve at 0x000001979C7E4AE0>
           │      └ <function Runner.run at 0x000001979BF64180>
           └ <asyncio.runners.Runner object at 0x000001979A2BFAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
           │    │     │                  └ <Task pending name='Task-1' coro=<Server.serve() running at C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-p...
           │    │     └ <function BaseEventLoop.run_until_complete at 0x000001979BF5DC60>
           │    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
           └ <asyncio.runners.Runner object at 0x000001979A2BFAD0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 640, in run_until_complete
    self.run_forever()
    │    └ <function BaseEventLoop.run_forever at 0x000001979BF5DBC0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 607, in run_forever
    self._run_once()
    │    └ <function BaseEventLoop._run_once at 0x000001979BF5F9C0>
    └ <_WindowsSelectorEventLoop running=True closed=False debug=False>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\base_events.py", line 1922, in _run_once
    handle._run()
    │      └ <function Handle._run at 0x000001979BE95260>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\asyncio\events.py", line 80, in _run
    self._context.run(self._callback, *self._args)
    │    │            │    │           │    └ <member '_args' of 'Handle' objects>
    │    │            │    │           └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    │            │    └ <member '_callback' of 'Handle' objects>
    │    │            └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
    │    └ <member '_context' of 'Handle' objects>
    └ <Handle Task.task_wakeup(<Future finis...rrent_user'")>)>
> File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\protocols\http\httptools_impl.py", line 426, in run_asgi
    result = await app(  # type: ignore[func-returns-value]
                   └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x000001979FBF2ED0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\uvicorn\middleware\proxy_headers.py", line 84, in __call__
    return await self.app(scope, receive, send)
                 │    │   │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x000001979F...
                 │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
                 │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
                 │    └ <fastapi.applications.FastAPI object at 0x000001979E90FE90>
                 └ <uvicorn.middleware.proxy_headers.ProxyHeadersMiddleware object at 0x000001979FBF2ED0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\applications.py", line 1106, in __call__
    await super().__call__(scope, receive, send)
                           │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x000001979F...
                           │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
                           └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\applications.py", line 122, in __call__
    await self.middleware_stack(scope, receive, send)
          │    │                │      │        └ <bound method RequestResponseCycle.send of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x000001979F...
          │    │                │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │                └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x000001979F3B0610>
          └ <fastapi.applications.FastAPI object at 0x000001979E90FE90>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 184, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\errors.py", line 162, in __call__
    await self.app(scope, receive, _send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x000001979FDED4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.cors.CORSMiddleware object at 0x000001979FC4E910>
          └ <starlette.middleware.errors.ServerErrorMiddleware object at 0x000001979F3B0610>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\cors.py", line 83, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ServerErrorMiddleware.__call__.<locals>._send at 0x000001979FDED4E0>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.sessions.SessionMiddleware object at 0x000001979FE10090>
          └ <starlette.middleware.cors.CORSMiddleware object at 0x000001979FC4E910>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\sessions.py", line 86, in __call__
    await self.app(scope, receive, send_wrapper)
          │    │   │      │        └ <function SessionMiddleware.__call__.<locals>.send_wrapper at 0x000001979FDEE980>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x000001979FE10950>
          └ <starlette.middleware.sessions.SessionMiddleware object at 0x000001979FE10090>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 79, in __call__
    raise exc
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\middleware\exceptions.py", line 68, in __call__
    await self.app(scope, receive, sender)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x000001979FE10C50>
          └ <starlette.middleware.exceptions.ExceptionMiddleware object at 0x000001979FE10950>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 20, in __call__
    raise e
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\middleware\asyncexitstack.py", line 17, in __call__
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <fastapi.routing.APIRouter object at 0x000001979C9BD650>
          └ <fastapi.middleware.asyncexitstack.AsyncExitStackMiddleware object at 0x000001979FE10C50>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 718, in __call__
    await route.handle(scope, receive, send)
          │     │      │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │     │      │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │     │      └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │     └ <function Route.handle at 0x000001979E88F600>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 276, in handle
    await self.app(scope, receive, send)
          │    │   │      │        └ <function ExceptionMiddleware.__call__.<locals>.sender at 0x000001979FDEEA20>
          │    │   │      └ <bound method RequestResponseCycle.receive of <uvicorn.protocols.http.httptools_impl.RequestResponseCycle object at 0x0000019...
          │    │   └ {'type': 'http', 'asgi': {'version': '3.0', 'spec_version': '2.3'}, 'http_version': '1.1', 'server': ('127.0.0.1', 8000), 'cl...
          │    └ <function request_response.<locals>.app at 0x000001979FD27E20>
          └ APIRoute(path='/categories', name='create_category', methods=['POST'])
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\routing.py", line 66, in app
    response = await func(request)
                     │    └ <starlette.requests.Request object at 0x000001979FE13E50>
                     └ <function get_request_handler.<locals>.app at 0x000001979FD25440>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 274, in app
    raw_response = await run_endpoint_function(
                         └ <function run_endpoint_function at 0x000001979E88D940>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\fastapi\routing.py", line 193, in run_endpoint_function
    return await run_in_threadpool(dependant.call, **values)
                 │                 │         │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
                 │                 │         └ <function create_category at 0x000001979FC66AC0>
                 │                 └ <fastapi.dependencies.models.Dependant object at 0x000001979FDEA750>
                 └ <function run_in_threadpool at 0x000001979CD580E0>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\starlette\concurrency.py", line 41, in run_in_threadpool
    return await anyio.to_thread.run_sync(func, *args)
                 │     │         │        │      └ ()
                 │     │         │        └ functools.partial(<function create_category at 0x000001979FC66AC0>, current_user=<User: User object>, create_category_use_cas...
                 │     │         └ <function run_sync at 0x000001979C741080>
                 │     └ <module 'anyio.to_thread' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\to_...
                 └ <module 'anyio' from 'C:\\Users\\tmcon\\AppData\\Local\\Programs\\Python\\Python311\\Lib\\site-packages\\anyio\\__init__.py'>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\to_thread.py", line 33, in run_sync
    return await get_asynclib().run_sync_in_worker_thread(
                 └ <function get_asynclib at 0x000001979C6F2200>
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 877, in run_sync_in_worker_thread
    return await future
                 └ <Future finished exception=TypeError("CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_...
  File "C:\Users\tmcon\AppData\Local\Programs\Python\Python311\Lib\site-packages\anyio\_backends\_asyncio.py", line 807, in run
    result = context.run(func, *args)
             │       │   │      └ ()
             │       │   └ functools.partial(<function create_category at 0x000001979FC66AC0>, current_user=<User: User object>, create_category_use_cas...
             │       └ <method 'run' of '_contextvars.Context' objects>
             └ <_contextvars.Context object at 0x000001979FF6F680>

  File "E:\spending-webapp\backend\app\shared\decorator.py", line 24, in wrapper
    response = f(*args, **kwargs)
               │  │       └ {'current_user': <User: User object>, 'create_category_use_case': <app.use_cases.category.create.CreateCategoryUseCase object...
               │  └ ()
               └ <function create_category at 0x000001979FC66A20>

  File "E:\spending-webapp\backend\app\interfaces\rest\api_v1\endpoints\category.py", line 44, in create_category
    req_object = CreateCategoryRequestObject.builder(payload=payload)
                 │                           │               └ CategoryInCreate(name='Lương tháng', type=<Type.INCOME: 'income'>, note='Dùng khi nhận lương')
                 │                           └ <classmethod(<function CreateCategoryRequestObject.builder at 0x000001979FC67920>)>
                 └ <class 'app.use_cases.category.create.CreateCategoryRequestObject'>

TypeError: CreateCategoryRequestObject.builder() missing 1 required positional argument: 'current_user'

2024-01-14 11:12:14.298 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:12:14.298 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:12:14.408 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:12:14.408 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:12:14.409 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:12:14.410 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:12:14.410 | INFO     | logging:callHandlers:1706 - Finished server process [11884]
2024-01-14 11:12:14.411 | INFO     | logging:callHandlers:1706 - Finished server process [11884]
2024-01-14 11:12:18.120 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:12:18.120 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:12:18.121 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:12:18.269 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:12:18.362 | INFO     | logging:callHandlers:1706 - Started server process [1392]
2024-01-14 11:12:18.362 | INFO     | logging:callHandlers:1706 - Started server process [1392]
2024-01-14 11:12:18.363 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:12:18.363 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:12:18.370 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:12:18.370 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:12:27.648 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:12:27.648 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:12:27.756 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:12:27.756 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:12:27.757 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:12:27.757 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:12:27.758 | INFO     | logging:callHandlers:1706 - Finished server process [1392]
2024-01-14 11:12:27.759 | INFO     | logging:callHandlers:1706 - Finished server process [1392]
2024-01-14 11:12:29.095 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:12:29.096 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:12:29.097 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:12:29.242 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:12:29.354 | INFO     | logging:callHandlers:1706 - Started server process [14892]
2024-01-14 11:12:29.355 | INFO     | logging:callHandlers:1706 - Started server process [14892]
2024-01-14 11:12:29.356 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:12:29.357 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:12:29.366 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:12:29.366 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:12:29.710 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52513 - "POST /categories HTTP/1.1" 500
2024-01-14 11:13:18.556 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:13:18.556 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:13:18.664 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:13:18.664 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:13:18.665 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:13:18.666 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:13:18.666 | INFO     | logging:callHandlers:1706 - Finished server process [14892]
2024-01-14 11:13:18.666 | INFO     | logging:callHandlers:1706 - Finished server process [14892]
2024-01-14 11:13:20.178 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'SafeConfigParser': <class 'configparser.ConfigParser'>
2024-01-14 11:13:20.179 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'NativeStringIO': <class '_io.StringIO'>
2024-01-14 11:13:20.179 | DEBUG    | logging:callHandlers:1706 - loaded lazy attr 'BytesIO': <class '_io.BytesIO'>
2024-01-14 11:13:20.365 | DEBUG    | logging:callHandlers:1706 - registered 'bcrypt' handler: <class 'passlib.handlers.bcrypt.bcrypt'>
2024-01-14 11:13:20.477 | INFO     | logging:callHandlers:1706 - Started server process [3160]
2024-01-14 11:13:20.478 | INFO     | logging:callHandlers:1706 - Started server process [3160]
2024-01-14 11:13:20.479 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:13:20.479 | INFO     | logging:callHandlers:1706 - Waiting for application startup.
2024-01-14 11:13:20.487 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:13:20.487 | INFO     | logging:callHandlers:1706 - Application startup complete.
2024-01-14 11:13:21.506 | INFO     | logging:callHandlers:1706 - 127.0.0.1:52524 - "POST /categories HTTP/1.1" 200
2024-01-14 11:13:28.681 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:13:28.681 | INFO     | logging:callHandlers:1706 - Shutting down
2024-01-14 11:13:28.788 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:13:28.788 | INFO     | logging:callHandlers:1706 - Waiting for application shutdown.
2024-01-14 11:13:28.790 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:13:28.790 | INFO     | logging:callHandlers:1706 - Application shutdown complete.
2024-01-14 11:13:28.790 | INFO     | logging:callHandlers:1706 - Finished server process [3160]
2024-01-14 11:13:28.791 | INFO     | logging:callHandlers:1706 - Finished server process [3160]
